appeals/web/src/server/appeals/appeal-details/appeal-timetables/appeal-timetables.controller.js

```diff
@@ -0,0 +1,182 @@
+import { appealShortReference } from '#lib/appeals-formatter.js';
+import { routeToObjectMapper, appealTimetablesMapper } from './appeal-timetables.mapper.js';
+
+/**
+ *
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ * @param {object} [apiErrors]
+ */
+const renderUpdateDueDate = async (request, response, apiErrors) => {
+	const appealDetails = request.currentAppeal;
+	const appealId = appealDetails.appealId;
+	const { timetableType } = request.params;
+
+	if (!appealId) {
+		return response.render('app/500.njk');
+	}
+
+	let errors = request.errors || apiErrors;
+
+	const timetableProperty = routeToObjectMapper[timetableType];
+
+	const { sideNote, page } = appealTimetablesMapper(
+		appealDetails?.appealTimetable,
+		timetableProperty
+	);
+
+	return response.render('appeals/appeal/update-due-date.njk', {
+		appeal: {
+			id: appealId,
+			shortReference: appealShortReference(appealDetails.appealReference)
+		},
+		sideNote,
+		page,
+		continueButtonText: 'Continue',
+		backButtonUrl: `/appeals-service/appeal-details/${appealId}`,
+		skipButtonUrl: null,
+		errors
+	});
+};
+
+/**
+ *
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+const processUpdateDueDate = async (request, response) => {
+	if (request.errors) {
+		return renderUpdateDueDate(request, response);
+	}
+
+	const { body } = request;
+	const appealDetails = request.currentAppeal;
+	const appealId = appealDetails.appealId;
+	const { appealTimetableId } = appealDetails.appealTimetable;
+	const { timetableType } = request.params;
+
+	if (!objectContainsAllKeys(body, ['due-date-day', 'due-date-month', 'due-date-year'])) {
+		return response.render('app/500.njk');
+	}
+
+	try {
+		const updatedDueDateDay = parseInt(body['due-date-day'], 10);
+		const updatedDueDateMonth = parseInt(body['due-date-month'], 10);
+		const updatedDueDateYear = parseInt(body['due-date-year'], 10);
+
+		if (
+			Number.isNaN(updatedDueDateDay) ||
+			Number.isNaN(updatedDueDateMonth) ||
+			Number.isNaN(updatedDueDateYear)
+		) {
+			return response.render('app/500.njk');
+		}
+
+		const updatedDueDateDayString = `0${updatedDueDateDay}`.slice(-2);
+		const updatedDueDateMonthString = `0${updatedDueDateMonth}`.slice(-2);
+
+		const timetableProperty = routeToObjectMapper[timetableType];
+
+		const setAppealTimetableResponse = await setAppealTimetables(
+			request.apiClient,
+			appealId,
+			appealTimetableId,
+			{
+				[timetableProperty]: `${updatedDueDateYear}-${updatedDueDateMonthString}-${updatedDueDateDayString}`
+			}
+		);
+
+		if (setAppealTimetableResponse.errors) {
+			const apiError = Object.values(setAppealTimetableResponse.errors)[0];
+			if (apiError) {
+				const apiErrors = {
+					'due-date-day': {
+						value: String(updatedDueDateDay),
```

Comment:
Should this handle the other date fields as well (month/year)? As far as I remember the API accepts and returns the date as a single string rather than the individual day/month/year parts, so this might not be possible.